andreeapetcu/assignment2/Ancestors.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * ASSUMPTIONS : the binary tree does not contain duplicate values
+ */
+
+import java.util.ArrayList;
+
+public class Ancestors<T> {
+
+  /**
+   * Wraper class that contains the ancestors of a node and a flag for the existence of a node
+   * with the given key.
+   */
+  class Result {
+    public ArrayList<T> ancestors;
+    public boolean flag;
```

Comment:
"flag" isn't a very descriptive name. Can you think of a better one? I'd probably use something like "valueFound".